Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/PredictorCorrector.java

```diff
@@ -705,10 +709,12 @@ protected <P extends AnomalyDescriptor> P detect(P result, RCFComputeDescriptor
         boolean candidate = false;
 
         if (workingGrade > 0 && lastDescriptor != null) {
-            if (score > lastDescriptor.getRCFScore()
-                    || lastDescriptor.getRCFScore() - lastDescriptor.getThreshold() > score
-                            - max(workingThreshold, lastDescriptor.getThreshold())
-                                    * (1 + max(0.2, runLength / (2.0 * max(10, shingleSize))))) {
+            if (score > lastDescriptor.getRCFScore()) {
+                candidate = true;
+            }
+            double runDiscount = max(workingThreshold, lastDescriptor.getThreshold())
+                    * (1 + max(0.2, runLength / (2.0 * max(10, shingleSize))));
+            if (lastDescriptor.getRCFScore() - lastDescriptor.getThreshold() > score - runDiscount) {
```

Comment:
Can you explain the intuition behind the condition? It looks like you are detecting decreasing gap by making the threshold larger and subtracting larger value. But the rationale is opaque to me.

Reply:
See above.